src/catalog/event-bus/src/test/java/org/geoserver/cloud/event/bus/BusEventCollector.java

```diff
@@ -92,6 +97,21 @@ public <T extends InfoEvent> RemoteGeoServerEvent expectOne(
         return matches.get(0);
     }
 
+    public <T extends LifecycleEvent> RemoteGeoServerEvent expectOneLifecycleEvent(
+            Class<T> payloadType) {
+
+        List<RemoteGeoServerEvent> matches =
+                await().atMost(Duration.ofSeconds(500)) //
```

Comment:
that's a rather long wait... I get you'd have set it up large so it gives you time to debug (I do the same), but in the end for github actions 10s should be plenty even if the build stalls. And yes, I've seen `expectOne(...)` already does that, my bad :/